echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/pagerduty/PagerDutyNotificationService.groovy

```diff
@@ -50,17 +58,91 @@ class PagerDutyNotificationService implements NotificationService {
 
   @Override
   void handle(Notification notification) {
+    Map details = new LinkedHashMap()
+    details.put("details", notification.additionalContext.details)
+
+    Map<String, String> errors = new HashMap()
+    Map<String, String> pdErrors = new HashMap()
+
     notification.to.each {
-      pagerDuty.createEvent(
-        "Token token=${token}",
-        new PagerDutyService.PagerDutyCreateEvent(
-          service_key: it,
-          client: "Spinnaker (${notification.source.user})",
-          description: notification.additionalContext.message
+      try {
+        Map response = pagerDuty.createEvent(
+          "Token token=${token}",
+          new PagerDutyService.PagerDutyCreateEvent(
+            service_key: it,
+            client: "Spinnaker (${notification.source.user})",
+            description: notification.additionalContext.message,
+            details: details
+          )
         )
-      )
 
-      log.info('Sent page {} {}', kv('serviceKey', it), kv('message', notification.additionalContext.message))
+        if (response.status.equals("success")) {
+          // Page successful
+          log.info('Sent page {} {}', kv('serviceKey', it), kv('message', notification.additionalContext.message))
+        } else {
+          pdErrors.put(it, (String)response.message)
+        }
+      } catch (RetrofitError error) {
+        String errorMessage = error.response.reason
+        TypedInput responseBody = error.response.getBody()
+        if (responseBody != null) {
+          PagerDutyErrorResponseBody errorResponse = mapper.readValue(new String(((TypedByteArray)responseBody).getBytes()), PagerDutyErrorResponseBody)
```

Comment:
IMO try and keep to 120 line width (RobF does 80 but that's _crazy_) ...

```
def errorResponse = mapper.readValue(
  new String(....),
  PagerDutyErrorResponseBody
)
```

is more readable IMO.